Eastern-Research-Group/expert-query
# Feature/289 approved users config

## Related Issues:
* [EQ-289](https://jira.epa.gov/browse/EQ-289)
* [EQ-290](https://jira.epa.gov/browse/EQ-290)

## Main Changes:
* Updated the `protectRoutes` middleware so that it checks the api.epa.gov api user id against a list of approved user ids. This code is only for the dev and stage environments. 
* Moved the tableConfig files to the s3 bucket. Now we just have one shared `tableConfig.json` file for both the api server and etl.
  * Note: These private bucket configs are only loaded once (the first time the `getPrivateConfig` function is called). This means that in cloud.gov we will have to restage the app when we change these files in s3. I think this is desirable, especially for the `tableConfig.json` which will affect the endpoints. 
* Upgraded to version 3 of the aws-sdk. We were getting warnings in the console when starting the server.
  * I manually tested this by having my local point to a test S3 bucket. 

## Steps To Test:
1. Run the etl
2. Verify it succeeds
3. Start the app
4. Verify all features still work

## TODO:
- [ ] After merging this in, we will need to test from the dev environment. Including, re-running the etl.
- [ ] Move more of the client configs to the public s3 bucket. This will probably be most items in the `app/client/src/config` folder. I'll create a separate ticket for this.
